src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -245,6 +273,21 @@ private UnboundLambda BindAnonymousFunction(CSharpSyntaxNode syntax, DiagnosticB
                         continue;
                     }
 
+                    if (lambda.ParameterIsDiscard(i))
+                    {
+                        if (seenDiscard)
+                        {
+                            // We only report the diagnostic on the second and subsequent underscores
+                            MessageID.IDS_FeatureLambdaDiscardParameters.CheckFeatureAvailability(
+                                diagnostics,
+                                binder.Compilation,
+                                lambda.ParameterLocation(i));
+                        }
+
+                        seenDiscard = true;
+                        continue;
```

Comment:
>continue; [](start = 24, length = 9)

Consider adding a test that would verify that we don't get here for a regular parameter named ```_```. I.e. we are still going to report a diagnostics from ```ValidateLambdaParameterNameConflictsInScope``` when it is appropriate.

Reply:
Added test `DiscardParameters_SingleUnderscoreParameter`

---
In reply to: [340741002](https://github.com/dotnet/roslyn/pull/38786#discussion_r340741002) [](ancestors = 340741002)

Reply:
> Added test DiscardParameters_SingleUnderscoreParameter

I doesn't look like the test is covering the scenario. I would expect an error about parameter having a conflicting name with something in an outer scope. The test is asserting a conflict of something declared in a nested scope with a parameter instead. Therefore, doesn't test the code path.

---
In reply to: [341279707](https://github.com/dotnet/roslyn/pull/38786#discussion_r341279707) [](ancestors = 341279707,340741002)

Reply:
I'm adding another test, but this `ValidateLambdaParameterNameConflictsInScope` won't be reachable. `allowShadowingNames` is true as of C# 8.

---
In reply to: [341297823](https://github.com/dotnet/roslyn/pull/38786#discussion_r341297823) [](ancestors = 341297823,341279707,340741002)